pkg/apis/core/validation/validation.go

```diff
@@ -4175,10 +4175,7 @@ func validateServiceExternalTrafficFieldsValue(service *core.Service) field.Erro
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("externalTrafficPolicy"), service.Spec.ExternalTrafficPolicy,
 			fmt.Sprintf("ExternalTrafficPolicy must be empty, %v or %v", core.ServiceExternalTrafficPolicyTypeCluster, core.ServiceExternalTrafficPolicyTypeLocal)))
 	}
-	// 'externalTrafficPolicy=Local' is mutually exclusive with topologyKeys
-	if service.Spec.ExternalTrafficPolicy == core.ServiceExternalTrafficPolicyTypeLocal && len(service.Spec.TopologyKeys) > 0 {
```

Comment:
Other validation check is here https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L4064